File: prototype/rails/src/test/domain/shop/status_label_test.rb
require "test_helper"

class StatusLabelTest < ActiveSupport::TestCase
  StatusLabel = Domain::Shop::StatusLabel

  test "it reads a stored status as a sentence" do
    assert_equal "Awaiting shipment", StatusLabel.humanize("awaiting_shipment")
  end

  test "a single word status keeps its shape" do
    assert_equal "Paid", StatusLabel.humanize("paid")
  end
end

File: prototype/node/src/app/core/shop/status-label.test.ts
import { test } from 'node:test'
import assert from 'node:assert/strict'
import { statusLabel } from './status-label.ts'

test('it reads a stored status as a sentence', () => {
  assert.equal(statusLabel('awaiting_shipment'), 'Awaiting shipment')
})

test('a single word status keeps its shape', () => {
  assert.equal(statusLabel('paid'), 'Paid')
})

test('an empty status reads as nothing', () => {
  assert.equal(statusLabel(''), '')
})
